modules/ipi-install-configuring-storage-on-nodes.adoc

```diff
@@ -0,0 +1,52 @@
+// Module included in the following assemblies:
+//
+// * list of assemblies where this module is included
+// ipi-install-installation-workflow.adoc
+
+:_content-type: PROCEDURE
+[id="configuring-storage-on-nodes_{context}"]
+= Optional: Configuring storage on nodes
+
+Changes to operating systems on {product-title} nodes can be done by creating what are referred to as `MachineConfig` objects that are managed by the Machine Config Operator (MCO).
+
+The `MachineConfig` specification includes an ignition config for configuring the machines at first boot. This config object can be used to do things like modify files, systemd services, and other operating system features running on on {product-title} machines.
+
+Use the following procedure to configure storage on a new node.
+
+.Procedure
+
+The ignition config can be used to configure storage on the nodes. The following `MachineSet` manifest example demonstrates how to add a partition to sda device on master nodes. I.e. apply the manifest before installation to have a partition named recovery of size 16 GiB on the master nodes.
+
+. Create the new `MachineConfig` manifest:
++
+[source,terminal]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 10_masters_storage_config
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      disks:
+        - device: /dev/sda
+          partitions:
+            - label: recovery
+              startMiB: 32768
+              sizeMiB: 16384
+      filesystems:
+        - device: /dev/disk/by-partlabel/recovery
+          label: recovery
+          format: xfs
+----
++
+. Save and copy the new `MachineConfig` manifest to the `clusterconfigs/openshift` directory:
++
+[source,terminal]
+----
+cp ~/<MachineConfig_manifest> clusterconfigs/openshift
```

Comment:
Hi @danielerez -- So my intention here was to adapt this content into a generic OCP use case of configuring partitions on installation. If this functionality is verified and support, I think it could be a benefit to all environments, even ones not deployed with AI. If this is something you don't think should be documented as a generic OCP use case, then we can close this PR.

That said, is it possible to perform this functionality with Assisted Installer SaaS? Would we need to add a section in the AI SaaS doc's [Installing with the Assisted Installer API](https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2022/html/assisted_installer_for_openshift_container_platform/installing-with-api#doc-wrapper) section instead?

Reply:
Oh ok, I've just assumed the context here is AI. For a generic use-case, this example indeed makes sense.